src/disaster_vehicle.cpp

```diff
@@ -578,17 +579,27 @@ static bool DisasterTick_Big_Ufo(DisasterVehicle *v)
 		}
 		v->current_order.SetDestination(1);
 
-		TileIndex tile_org = RandomTile();
-		TileIndex tile = tile_org;
-		do {
-			if (IsPlainRailTile(tile) &&
-					Company::IsHumanID(GetTileOwner(tile))) {
+		uint n = 0; // Total number of targetable trains.
+		for (const Train *t : Train::Iterate()) {
+			if (t->IsFrontEngine()) n++;
+		}
+
+		if (n == 0) {
+			/* If there are no targetable trains, destroy the UFO. */
+			delete v;
+			return false;
+		}
+
+		n = RandomRange(n); // Choose one of them.
+		for (const Train *t : Train::Iterate()) {
+			/* Find (n+1)-th road vehicle. */
```

Comment:
```suggestion
			/* Find (n+1)-th train. */
```

Reply:
Can you tell the code was copied?

Reply:
I actually read it in the description. :)